packages/app/src/overview/components/overview-screens/helpers/use-update-flow-state.hook.tsx

```diff
@@ -3,85 +3,139 @@
+      !needsForceUpdate &&
       osVersion
     ) {
-      const actionResult = checkForUpdate()
+      const actionResult = silentCheck()
       setSilentCheckForUpdatePromise(actionResult)
     }
 
     return () => {
       if (
         silentCheckForUpdatePromise &&
-        silentCheckForUpdateState === SilentCheckForUpdateState.Loading
+        silentCheckForUpdate === SilentCheckForUpdateState.Loading
       ) {
         silentCheckForUpdatePromise.abort()
         setSilentCheckForUpdatePromise(undefined)
       }
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [silentCheckForUpdateState, forceUpdateNeeded, osVersion])
+  }, [silentCheckForUpdate, needsForceUpdate, osVersion])
 
   useEffect(() => {
-    if (forceUpdateNeeded) {
+    if (needsForceUpdate) {
       return
     }
     if (
-      (silentCheckForUpdateState === SilentCheckForUpdateState.Skipped &&
-        checkingForUpdateState === CheckForUpdateState.Initial) ||
-      silentCheckForUpdateState === SilentCheckForUpdateState.Failed ||
-      checkingForUpdateState === CheckForUpdateState.Failed ||
-      checkingForUpdateState === CheckForUpdateState.PerformedWithFailure
+      silentCheckForUpdate === SilentCheckForUpdateState.Loading ||
+      checkForUpdateState === CheckForUpdateState.Loading ||
+      (checkForUpdateState === CheckForUpdateState.Initial &&
+        silentCheckForUpdate === SilentCheckForUpdateState.Initial)
     ) {
-      setCheckForUpdateLocalState(CheckForUpdateLocalState.Failed)
+      setCheckForUpdateLocalState(CheckForUpdateLocalState.Loading)
+    } else if (allDownloaded) {
+      setCheckForUpdateLocalState(CheckForUpdateLocalState.Install)
+    } else if ((availableReleasesForUpdate || []).length > 0) {
+      setCheckForUpdateLocalState(CheckForUpdateLocalState.Download)
     } else if (
-      silentCheckForUpdateState === SilentCheckForUpdateState.Loading
+      error ||
+      [
+        CheckForUpdateState.PerformedWithFailure,
+        CheckForUpdateState.Failed,
+      ].includes(checkForUpdateState) ||
+      (checkForUpdateState === CheckForUpdateState.Initial &&
+        [
+          SilentCheckForUpdateState.Skipped,
+          SilentCheckForUpdateState.Failed,
+        ].includes(silentCheckForUpdate))
     ) {
-      setCheckForUpdateLocalState(CheckForUpdateLocalState.SilentCheckLoading)
-    } else if (checkingForUpdateState === CheckForUpdateState.Loading) {
-      setCheckForUpdateLocalState(CheckForUpdateLocalState.Loading)
+      setCheckForUpdateLocalState(CheckForUpdateLocalState.Failed)
     } else if (
-      silentCheckForUpdateState === SilentCheckForUpdateState.Loaded ||
-      checkingForUpdateState === CheckForUpdateState.Loaded ||
-      checkingForUpdateState === CheckForUpdateState.Performed
+      [CheckForUpdateState.Loaded, CheckForUpdateState.Performed].includes(
+        checkForUpdateState
+      ) ||
+      SilentCheckForUpdateState.Loaded === silentCheckForUpdate
```

Comment:
Those would have to be 5 different functions with different condition each or another hook inside a hook that exist solely to wrap 90% of the outer hooks logic. To be fair I don't think I can name those conditions descriptively.

Reply:
I understand it. Thank you for considering it. These approaches require additional effort. I do not intend to push this topic. It can stay as it is.